src/pages/resultsView/fusion/ResultViewFusionMapperStore.ts

```diff
@@ -0,0 +1,137 @@
+
+    constructor(public gene: Gene,
+                public studyIdToStudy: MobxPromise<{ [studyId: string]: CancerStudy }>,
+                public molecularProfileIdToMolecularProfile: MobxPromise<{ [molecularProfileId: string]: MolecularProfile }>,
+                public samples: MobxPromise<SampleIdentifier[]>,
+                public fusions: StructuralVariantExt[]) {
+        labelMobxPromises(this);
+    }
+
+    private applyColor (result: EnsemblTranscriptExt[]) {
+        let fusionTranscripts: EnsemblTranscriptExt[] = [];
+        return result.map(t => {
+            // Check if transcript is reference gene, if yes then mark it
+            t.isReferenceGene = t.hugoSymbols.indexOf(this.gene.hugoGeneSymbol) >= 0;
+            if (t.isReferenceGene) {
+                t.fillColor = REF_GENE_COLORSCALE[0];
+            } else {
+                t.fillColor = FUS_GENE_COLORSCALE[fusionTranscripts.length % FUS_GENE_COLORSCALE.length];
+                const countRef = fusionTranscripts.filter(ref => {
+                    return ref['geneId'] === t['geneId']
+                });
+                if (countRef.length === 0) {
+                    fusionTranscripts.push(t)
+                }
+            }
+            return t;
+        });
+    };
+
+    readonly ensemblTranscripts = remoteData({
+        invoke: async () => {
+            // Collect all transcript id's from the fusion data
+            let _transcriptIds = new Set();
+            this.fusions.forEach((fusion: StructuralVariant) => {
+                _transcriptIds.add(fusion.site1EnsemblTranscriptId);
+                _transcriptIds.add(fusion.site2EnsemblTranscriptId);
+            });
+            let transcriptIds: string[] = Array.from(_transcriptIds);
+            // Fetch ensembl transcript data with transcript ids as request params
+            return await fetchEnsemblTranscriptsByEnsemblFilter({'transcriptIds': transcriptIds})
+                .then( result => this.applyColor(result));
+        }
+    }, []);
+
+    get exonsChartStore(): ExonsChartStore {
+        return new ExonsChartStore(
+            this.gene,
+            this.ensemblTranscripts,
+            this.pfamDomainData,
+            this.fusions
+        )
+    }
+
+    readonly pfamDomainData = remoteData<PfamDomain[] | undefined>({
+        await: () => [
+            this.ensemblTranscripts
+        ],
+        invoke: () => {
+            // get all pfam domain ids from requested transcripts
+            let pFamDomainIds = this.ensemblTranscripts.result
+                .map(t => t.pfamDomains)
+                .reduce((accumulator, currentValue) => accumulator.concat(currentValue), [])
+                .map(t => t.pfamDomainId);
+            // fetch the data with pfam domain ids as request params
+            return fetchPfamDomainData(pFamDomainIds);
+        }
+    }, []);
+
+    @computed
+    get fusionOccurrence(): { [fusionLabel: string]: number } {
```

Comment:
I still think `fusionCounts` is a better name. Occurence is vague to me, unless thats a biological term I don't know. what do you think?

Reply:
sounds good